climada/util/lines_polys_handler.py

```diff
@@ -0,0 +1,891 @@
+    containing the geometries.
+
+    Parameters
+    ----------
+    impact_pnt : Impact
+        Impact object with impact per exposure point (lines of exp_pnt)
+    exp_pnt : Exposures
+        Exposures with a double index geodataframe, first for geometries,
+        second for the point disaggregation of the geometries.
+    agg : string, optional
+        If 'agg', the impact is averaged over all points in each geometry.
+        If 'sum', the impact is summed over all points in each geometry.
+        The default is 'sum'.
+
+    Returns
+    -------
+    Impact
+        Impact object with the impact per geometry.
+
+    """
+
+    # aggregate impact
+    mat_agg = aggregate_impact_mat(impact_pnt, exp_pnt.gdf, agg)
+
+    #Write to impact obj
+    impact_agg = set_imp_mat(impact_pnt, mat_agg)
+
+    #Add exposure representation points as coordinates
+    repr_pnts = gpd.GeoSeries(
+        exp_pnt.gdf['geometry_orig'][:,0].apply(
+            lambda x: x.representative_point()
+            )
+        )
+    impact_agg.coord_exp = np.array([repr_pnts.y, repr_pnts.x]).transpose()
+    #Add geometries
+    impact_agg.geom_exp = exp_pnt.gdf.xs(0, level=1)\
+        .set_geometry('geometry_orig')\
+            .geometry.rename('geometry')
+
+    return impact_agg
+
+def aggregate_impact_mat(imp_pnt, gdf_pnt, agg):
+    """
+    Aggregate impact matrix given geodataframe or disaggregated polygons.
+
+    Parameters
+    ----------
+    impact_pnt : Impact
+        Impact object with impact per point (lines of gdf_pnt)
+    gdf_pnt : GeoDataFrame
+        Exposures geodataframe with a double index, first for polygon geometries,
+        second for the point disaggregation of the polygons.
+    agg : string
+        If 'agg', the impact is averaged over all points in each polygon.
+        If 'sum', the impact is summed over all points in each polygon.
+
+    Returns
+    -------
+    sparse.csr_matrix
+        matrix num_events x num_polygons with impacts.
+
+    """
+    # aggregate impact
+    mi = gdf_pnt.index
+    col_geom = mi.get_level_values(level=0)
+    def convert_str_idx_to_int(idx):
+        """Converts string multi-index level 0 to integer index"""
+        return np.sort(np.unique(idx, return_inverse=True)[1])
+    col_geom = convert_str_idx_to_int(col_geom)
+    row_pnt = np.arange(len(col_geom))
+    if agg == 'avg':
+        from collections import Counter
+        geom_sizes = Counter(col_geom).values()
+        mask = np.concatenate([np.ones(l) / l for l in geom_sizes])
+    elif agg == 'sum':
+        mask = np.ones(len(col_geom))
+    else:
+        raise ValueError("Please choose a valid aggregation method. agg=%s is not valid", agg)
+    csr_mask = sp.sparse.csr_matrix(
+        (mask, (row_pnt, col_geom)),
+         shape=(len(row_pnt), len(np.unique(col_geom)))
+        )
+    return imp_pnt.imp_mat.dot(csr_mask)
+
+def plot_eai_exp_geom(imp_geom, centered=False, figsize=(9, 13), **kwargs):
```

Comment:
can we extend this to a general `plot_impact_geom` function where we have as a kwarg the different usual impact variable options `eai`, `at_event`, `aai`, etc.? 
Especially for a single event calculation, I don't like the frequent "misuse" of eai (being the expected _annual_ impact) to plot the event impact.

Reply:
I guess so yes, good idea!

Reply:
The cleanest would probably be to make a sub-class `Impact_geom` or similar with the extra methods and attributes.

Reply:
Keep this as an extension idea in the issues section.